### DISCUSSION TITLE
来升级到 antd@5 吧

### DISCUSSION THREAD

**Discussion by chenshuai2144** [ant-design/ant-design#36814] [Q&A]
这里可以写所有升级antd@5 碰到的问题 或者 需求

**Comment by chenshuai2144** [comment]
<img width="451" alt="image" src="https://user-images.githubusercontent.com/8186664/182064798-5b0d4dc6-827a-4b5a-b784-82a8b4a66899.png">


antd@4 也需要一个 useToken 来帮助组件库做到互相兼容

具体方案

https://github.com/react-component/util/blob/master/src/hooks/useId.ts#L3

**Reply by spt110** [reply]
formily-antd-v5升级说明
1.以下升级到最新:
"react": "^18.2.0",
"react-dom": "^18.2.0",
"@formily/antd-v5": "1.0.1-rc.1",
"@formily/core": "^2.2.19",
"@formily/json-schema": "^2.2.19",
"@formily/path": "^2.2.19",
"@formily/react": "^2.2.19",
"@formily/reactive": "^2.2.19",
"@formily/reactive-react": "^2.2.19",
"@formily/shared": "^2.2.19",

2.添加安装
"@ant-design/antd-theme-variable": "^1.0.0",
 "rc-util": "^5.28.0",

3.umi4 添加配置
import theme from '@ant-design/antd-theme-variable';
export default {
  npmClient: 'pnpm',
  theme: {
    '@ant-prefix': 'ant',
    ...theme,
    '@s-site-menu-width': '258px',
    '@root-entry-name': 'variable',
    '@border-radius-base': '4px',
    '@primary-color': '#1677FF',
    '@error-color': '#ff4d4',
    '@info-color': '#52c41a',
    '@warning-color': '#faad14',
  },
   alias: {
       '@formily/antd': '@formily/antd-v5',
   }

4.umi4 删除配置 babel-plugin-import
export default {
 extraBabelPlugins: [
    // [
    //   'babel-plugin-import',
    //   { libraryName: 'antd', libraryDirectory: 'es', style: true },
    //   'antd',
    // ],
    // [
    //   'babel-plugin-import',
    //   { libraryName: '@formily/antd-v5', libraryDirectory: 'esm', style: true },
    //   '@formily/antd',
    // ],
  ],
  }
5. 删除所有  antd/es/style/themes/index.less'
//@import (reference) '~antd/es/style/themes/index.less';

**Reply by qingbopro** [reply]
没有生效....

// https://umijs.org/config/
import theme from '@ant-design/antd-theme-variable';
import { defineConfig } from '@umijs/max';
import { join } from 'path';

import proxy from './proxy';
import routes from './routes';

const { REACT_APP_ENV = 'dev' } = process.env;
const config: ReturnType<typeof defineConfig> = defineConfig({
  /**
   * @name 开启 hash 模式
   * @description 让 build 之后的产物包含 hash 后缀。通常用于增量发布和避免浏览器加载缓存。
   * @doc https://umijs.org/docs/api/config#hash
   */
  hash: true,
  favicons: ['/favicon.png'],
  tailwindcss: {},
  /**
   * @name 兼容性设置
   * @description 设置 ie11 不一定完美兼容，需要检查自己使用的所有依赖
   * @doc https://umijs.org/docs/api/config#targets
   */
  // targets: {
  //   ie: 11,
  // },
  /**
   * @name 路由的配置，不在路由中引入的文件不会编译
   * @description 只支持 path，component，routes，redirect，wrappers，title 的配置
   * @doc https://umijs.org/docs/guides/routes
   */
  // umi routes: https://umijs.org/docs/routing
  routes,
  /**
   * @name 主题的配置
   * @description 虽然叫主题，但是其实只是 less 的变量设置
   * @doc antd的主题设置 https://ant.design/docs/react/customize-theme-cn
   * @doc umi 的theme 配置 https://umijs.org/docs/api/config#theme
   */
  // theme: {
  //   // 如果不想要 configProvide 动态设置主题需要把这个设置为 default
  //   // 只有设置为 variable， 才能使用 configProvide 动态设置主色调
  //   // 'root-entry-name': 'variable',
  //   // theme: { '@primary-color': '#9e1082' },
  // },
  // lessLoader: {
  //   modifyVars: {
  //     '@primary-color': '#9e1082',
  //   },
  //   javascriptEnabled: true,
  // },
  /**
   * @name moment 的国际化配置
   * @description 如果对国际化没有要求，打开之后能减少js的包大小
   * @doc https://umijs.org/docs/api/config#ignoremomentlocale
   */
  ignoreMomentLocale: true,
  /**
   * @name 代理配置
   * @description 可以让你的本地服务器代理到你的服务器上，这样你就可以访问服务器的数据了
   * @see 要注意以下 代理只能在本地开发时使用，build 之后就无法使用了。
   * @doc 代理介绍 https://umijs.org/docs/guides/proxy
   * @doc 代理配置 https://umijs.org/docs/api/config#proxy
   */
  proxy: proxy[REACT_APP_ENV as keyof typeof proxy],
  /**
   * @name 快速热更新配置
   * @description 一个不错的热更新组件，更新时可以保留 state
   */
  fastRefresh: true,
  //============== 以下都是max的插件配置 ===============
  /**
   * @name 数据流插件
   * @@doc https://umijs.org/docs/max/data-flow
   */
  model: {},
  /**
   * 一个全局的初始数据流，可以用它在插件之间共享数据
   * @description 可以用来存放一些全局的数据，比如用户信息，或者一些全局的状态，全局初始状态在整个 Umi 项目的最开始创建。
   * @doc https://umijs.org/docs/max/data-flow#%E5%85%A8%E5%B1%80%E5%88%9D%E5%A7%8B%E7%8A%B6%E6%80%81
   */
  initialState: {},
  /**
   * @name layout 插件
   * @doc https://umijs.org/docs/max/layout-menu
   */
  title: '易数',
  layout: {
    locale: false,
  },
  /**
   * @name moment2dayjs 插件
   * @description 将项目中的 moment 替换为 dayjs
   * @doc https://umijs.org/docs/max/moment2dayjs
   */
  moment2dayjs: {
    preset: 'antd',
    plugins: ['duration'],
  },
  /**
   * @name antd 插件
   * @description 内置了 babel import 插件
   * @doc https://umijs.org/docs/max/antd#antd
   */
  // antd: {
  //   theme: {
  //     token: {
  //       colorPrimary: '#9e1082',
  //     },
  //   },
  // },
  // theme: {
  //   // '@ant-prefix': 'ant',
  //   // ...theme,
  //   // '@s-site-menu-width': '258px',
  //   // '@root-entry-name': 'variable',
  //   // '@border-radius-base': '4px',
  //   '@primary-color': '#9e1082',
  //   // '@error-color': '#ff4d4',
  //   // '@info-color': '#52c41a',
  //   // '@warning-color': '#faad14',
  // },
  /**
   * @name 网络请求配置
   * @description 它基于 axios 和 ahooks 的 useRequest 提供了一套统一的网络请求和错误处理方案。
   * @doc https://umijs.org/docs/max/request
   */
  request: {
    dataField: 'data',
  },
  /**
   * @name 权限插件
   * @description 基于 initialState 的权限插件，必须先打开 initialState
   * @doc https://umijs.org/docs/max/access
   */
  access: {},
  /**
   * @name <head> 中额外的 script
   * @description 配置 <head> 中额外的 script
   */
  headScripts: [
    // 解决首次加载时白屏的问题
    // {
    //   src: '/scripts/loading.js',
    //   async: true,
    // },
  ],
  theme: {
    '@ant-prefix': 'ant',
    ...theme,
    '@s-site-menu-width': '258px',
    '@root-entry-name': 'variable',
    '@border-radius-base': '4px',
    '@primary-color': '#9e1082',
    '@error-color': '#ff4d4',
    '@info-color': '#52c41a',
    '@warning-color': '#faad14',
  },
  alias: {
    '@formily/antd': '@formily/antd-v5',
  },
  //================ pro 插件配置 =================
  presets: ['umi-presets-pro'],
  mock: {
    include: ['mock/**/*', 'src/pages/**/_mock.ts'],
  },
  mfsu: {
    strategy: 'normal',
  },
  esbuildMinifyIIFE: true,
  requestRecord: {},
  codeSplitting: {
    jsStrategy: 'granularChunks',
  },
});
export default config;






"@formily/antd-v5": "^1.2.2",
"antd": "^5.19.1",
"@umijs/max": "^4.1.0",

**Comment by chenshuai2144** [comment]
如果项目中写了很多的less 变量，项目是无法跑起来的，这里有个兼容方案

```
const theme = require('@ant-design/antd-theme-variable');

```

删除 less  引用
```diff
- @import (reference) '~antd/es/style/themes/index.less';
- @import (reference) '~antd/es/style/mixins/index.less';
```

在 umi 中

```diff
 theme: {
    '@ant-prefix': 'ant',
+   ...theme,
    '@s-site-menu-width': '258px',
    '@root-entry-name': 'variable',
    '@border-radius-base': '4px',
    '@primary-color': '#1677FF',
    '@error-color': '#ff4d4',
    '@info-color': '#52c41a',
    '@warning-color': '#faad14',
  },
```

在 webpack 中

```diff
{
  loader: "less-loader",
  options: {
    lessOptions: {
      javascriptEnabled: true,
+   modifyVars: { ...theme, "@warning-color": "#faad14" },
    },
  },
};
```

**Reply by riverdone** [reply]
<img width="1134" alt="image" src="https://user-images.githubusercontent.com/51936110/204751134-0904384e-f88f-4761-9e14-a2264f95ada9.png">
umi项目启动还是报less错误。
下面是umi配置文件 config/config.ts
<img width="1655" alt="image" src="https://user-images.githubusercontent.com/51936110/204752381-e69edce0-9ff0-47da-ace9-53c84c6ae684.png">
package.json文件：
<img width="586" alt="image" src="https://user-images.githubusercontent.com/51936110/204752662-b33dbaa0-c93a-423e-a7f0-7a84e16fc75c.png">
请您帮我看下怎么解决 @chenshuai2144

**Reply by ByYogi** [reply]
请问您这个`theme`是从哪来的
![image](https://user-images.githubusercontent.com/111735598/205617217-6b3be2f7-a72b-46e0-b0f9-6668f5adfd99.png)
``` jsx
import { theme } from 'antd';
theme: {
      ...theme,
  },
```
``` jsx
.container {
  background: @layout-body-background;
}
````

**Reply by ByYogi** [reply]
> umi项目启动还是报less错误
> 请您帮我看下怎么解决 @chenshuai2144

兄弟，找了半天解决了。原来是没安装less-loader
相关链接：
https://webpack.docschina.org/loaders/less-loader/
https://blog.csdn.net/fifteen718/article/details/85704899
查看antd样式变量名: https://ant.design/docs/react/customize-theme-cn#seedtoken

相关代码
config.ts
``` tsx
import { theme } from 'antd';
const { defaultAlgorithm, defaultSeed } = theme;
const mapToken = defaultAlgorithm(defaultSeed);
export default defineConfig({
 lessLoader: {
    modifyVars: mapToken
  },
});
```
.less
``` diff
.div{
-  color: @layout-body-background;
-  background: @primary-color;
+  color: @colorPrimary;
+  background: @colorBgBase;
}
```

**Reply by sanheart** [reply]
> > umi项目启动还是报less错误
> > 请您帮我看下怎么解决 @chenshuai2144
> 
> 兄弟，找了半天解决了。原来是没安装less-loader 相关链接： https://webpack.docschina.org/loaders/less-loader/ https://blog.csdn.net/fifteen718/article/details/85704899 查看antd样式变量名: https://ant.design/docs/react/customize-theme-cn#seedtoken
> 
> 相关代码 config.ts
> 
> ```tsx
> import { theme } from 'antd';
> const { defaultAlgorithm, defaultSeed } = theme;
> const mapToken = defaultAlgorithm(defaultSeed);
> export default defineConfig({
>  lessLoader: {
>     modifyVars: mapToken
>   },
> });
> ```
> 
> .less
> 
> ```diff
> .div{
> -  color: @layout-body-background;
> -  background: @primary-color;
> +  color: @colorPrimary;
> +  background: @colorBgBase;
> }
> ```

您好。那么如何修改  @colorPrimary 的颜色呢

**Reply by ByYogi** [reply]
> 您好。那么如何修改 @colorPrimary 的颜色呢

你好，在umi中的配置文件中用下面配置
``` jsx
import { theme } from 'antd';

const { defaultAlgorithm, defaultSeed } = theme;
defaultSeed.colorPrimary = "red"; //这里修改自定义静态css变量
const mapToken = defaultAlgorithm(defaultSeed);

export default defineConfig({
  //配置这个属性需要安装  less-loader 插件
  //npm install less less-loader --save-dev
  lessLoader: {
    modifyVars: mapToken
  },
})
```

**Reply by wzlwangzhilin** [reply]
这个问题解决了吗？

**Reply by hjane** [reply]
但是我感觉这个方案想要实现动态换肤还是不行
没办法同步项目中动态改变的colorPrimary值和主题 light和dark值。

**Comment by xrkffgg** [comment]
顶一个

**Comment by afc163** [comment]
移除 babel-plugin-import 应该算一项。

**Reply by chenshuai2144** [reply]
umi 的插件现在有配置，你是希望默认配置吗

**Reply by afc163** [reply]
我希望后续默认关掉这个。

**Reply by chenshuai2144** [reply]
umi@3 可以先判断是antd@5 就关闭掉。
默认关最好等几个月

**Comment by Chastrlove** [comment]
1.next.js 引入直接antd 报错
```javascript
import { ConfigProvider } from "antd";
```
<img width="809" alt="image" src="https://user-images.githubusercontent.com/17813559/182069006-83be2471-3670-44cf-897d-5671648a7109.png">

2.ssr 下无法引入
```javascript
import zhCN from "antd/es/locale/zh_CN";
```
<img width="604" alt="image" src="https://user-images.githubusercontent.com/17813559/182069272-39da5ca5-b8b4-45b5-ab9d-1a92bc9fb003.png">

都是一些服务端遇到的问题(⊙o⊙)…

**Reply by chunsch** [reply]
可以跟进这个 PR：[feat: support for UMD bundles to be used on the ssr by chunsch · Pull Request #257 · ant-design/antd-tools · GitHub](https://github.com/ant-design/antd-tools/pull/257) : D

**Comment by chenshuai2144** [comment]
message.warn 的兼容
```
/**
 * 兼容antd@4 和 antd@5 的warning
 * @param messageStr
 */
const warning = (messageStr: React.ReactNode) => {
  // @ts-ignore
  return (message.warn || message.warning)(messageStr);
};

```

**Comment by chenshuai2144** [comment]
## dayjs 升级事项

```diff
- import moment from 'moment';
+ import dayjs from 'dayjs';
```
 ### isMoment -> isDayjs

```diff
- moment.isMoment(text);
+ dayjs.isDayjs(text);
```


### fromNow

```diff
- import moment from 'moment';
- moment(text).fromNow()

+ import dayjs from 'dayjs';
+ import relativeTime from 'dayjs/plugin/relativeTime';
+ dayjs.extend(relativeTime);
+ dayjs(text).fromNow()
```

**Comment by chenshuai2144** [comment]
## PageHeader

antd 删除了PageHeader，可以从 procomponents 中获取

```diff
- import type  { PageHeaderProps } from 'antd';
- import { PageHeader } from 'antd';

+ import type { PageHeaderProps } from '@ant-design/pro-components';
+ import { PageHeader } from '@ant-design/pro-components';
```

**Comment by kaoding** [comment]
希望可以看到antd v5的官网，看一看v5到底长什么样子，怎么升级和使用它。

**Reply by AnteeHub** [reply]
同意，希望能够看到一个预览版本的官网。

**Reply by ratelgogo** [reply]
希望尽快出文档

**Reply by tamaka365** [reply]
[应该是这个吧](https://next.ant.design/index-cn)

**Comment by jk4235** [comment]
哪里能看到antd v5的 changeLog 和 roadMap 呢

**Reply by TinsFox** [reply]
https://github.com/ant-design/ant-design/issues/34087

**Comment by chinyifei** [comment]
其他组件中能不能加一个**“伸缩框 ResizeBox”**组件

**Reply by afc163** [reply]
你说的是 https://github.com/tomkp/react-split-pane ？

**Reply by alienzhangyw** [reply]
> 你说的是 https://github.com/tomkp/react-split-pane ？

就类似这个吧https://x6.antv.vision/zh/docs/api/ui/splitbox
配合AutoScrollBox使用

**Reply by chinyifei** [reply]
我最终选择了这个库 re-resizable

**Comment by F-happy** [comment]
5.0.0-alpha.2 版本的 Popover 组件 overlayInnerStyle 属性设置了没有效果额

**Comment by zhenzhenChange** [comment]
样式变量（`CssVar/LessVar/Token`）可以在文档中的`API`栏增加一个分栏展示么，每次需要修改的时候去查，都得全局搜主题文件里的变量，[参考 Vant](https://vant3.uihtm.com/#/zh-CN/toast#zhu-ti-ding-zhi)。

**Comment by ghost** [comment]
Updated figma components for Ant design 5? Any news about it?

**Reply by jaybowang** [reply]
Yes! We need v5 in Figma!

**Comment by hexiaomin496** [comment]
https://ant.design/docs/react/migration-v5-cn
我们使用的umi4+pro components v2.3.34，已经按照文档的升级步骤，在UMI4的config.ts中移除了babel。
我们也使用了less，但文档中提供的是在webpack config 配置进行迁移，如何在UMI4进行兼容less？

也尝试了将文档中的建议迁移less的方法，配置到UMI4的config.ts中的less loader方法里
但是在执行 pnpm start 命令时报错，并提示：**cannot use import statement outside a module** 

我的配置方法如下：
![image](https://user-images.githubusercontent.com/23149314/204256103-accb846e-e086-48be-872b-ad561b023a29.png)
![image](https://user-images.githubusercontent.com/23149314/204256125-ff07519d-1248-40b4-a5c2-1ebc0ab27498.png)
![image](https://user-images.githubusercontent.com/23149314/204256142-c1dc77a7-c6dc-47fa-9e11-f3dabcd05223.png)


是否有在UMI4中使用antd V5的文档？我们已经按照迁移文档要求迁移到antd V4.24.4

**Reply by liuxi0811** [reply]
后来解决了么

**Comment by JC-Ge** [comment]
官网提示 Antd 5.x 要移除 Tabs.TabPane 组件，但实际依然可以使用。将来 Antd 5.x 会移除 TabPane 么?
[Antd 5.x Tabs](https://ant.design/components/tabs-cn/)

**Reply by yoyo837** [reply]
最好不用, 不保证它的稳定.

**Reply by BARMPlus** [reply]
不建议使用，最好使用最新的方式

**Comment by freewind** [comment]
升级到antd5时遇到的一个问题，谢谢：
https://github.com/ant-design/ant-design/discussions/39409

**Comment by n0tKnow** [comment]
请问官网这个demo有代码示例吗？感觉这个脚手架很好用
![image](https://user-images.githubusercontent.com/63541900/208290487-0b0d4df1-7373-4967-9d2b-e93506793df9.png)

**Reply by yoyo837** [reply]
在仓库源码里面

**Comment by GirlGlitter** [comment]
successfully upgraded👍

**Reply by liuxi0811** [reply]
How to do it

**Reply by GirlGlitter** [reply]
> How to do it

https://ant.design/docs/react/migration-v5

**Comment by yangda666** [comment]
升级antd 5 在 umi中使配置 AntdMomentWebpackPlugin 后启动不起来了 
const AntdMomentWebpackPlugin = require('@ant-design/moment-webpack-plugin');
 chainWebpack: (memo) => {
     memo.plugin('antd-moment-webpack-plugin')
        .use(AntdMomentWebpackPlugin)
}

**Reply by yangda666** [reply]
请问这样配的有问题吗，为什么跑不起来呢

**Reply by Wxh16144** [reply]
能否提供一个稳定复现的最小 repo？

**Reply by yangda666** [reply]
https://github.com/yangda666/antd-umi3.git  我连夜给你弄了一个

**Comment by iAmber** [comment]
版本 5.1.2  使用watermark组件报错

提示 /node_modules/antd/es/watermark/style/index

Does the file exist?

**Reply by Wxh16144** [reply]
请检查项目中是否使用了 `babel-plugin-import` 插件，5.x 已经不需要了。 

link: https://ant.design/docs/react/migration-v-5#remove-babel-plugin-import

**Reply by iAmber** [reply]
并没有使用 babel-plugin-import

**Reply by iAmber** [reply]
供参考，react + vite,  项目使用了 vite-plugin-style-import，移除即可

**Comment by popmanhe** [comment]
antd v5 SSR Theme的问题。

安装示例里面：
```
  const cache = createCache();

  const html = renderToString(
    <StyleProvider cache={cache}>
      <MyApp />
    </StyleProvider>,
  );


  // Grab style from cache
  const styleText = extractStyle(cache);
```
styleText 是空的。 我觉得是因为我的组件里面没有用cssinjs的方式。
但是如何获得antd里面的样式呢？
下图是客户端产生的样式。 
![image](https://user-images.githubusercontent.com/2193792/213799334-f31945ea-f237-477d-ad27-b1244abfa659.png)
下图是服务器端产生的
![image](https://user-images.githubusercontent.com/2193792/213800623-df8e1af3-500e-49f4-a416-a75fad8260a3.png)

服务器端却无法获得同样的样式。 这样服务器产生的页面就没有样式，而且会导致整个页面重新刷新一次。

**Comment by egene-huang** [comment]
请大佬们帮忙看一下这个问题： https://github.com/ant-design/ant-design/discussions/40442  
一直没有找到解决的办法，请救救这个孩子~

**Comment by yaojing-pan** [comment]
请问v5用定制主题怎么全局改button的font-weight，我没有找到对应的属性

**Comment by zhangqhzz** [comment]
我更新V5后，发现原来的一些样式有变化，影响到我的应用，如V4中有global.less 对HTML的P，h1,h2……h6元素有 margin-top: 0;这个相当于默认的，但在V5没有了，global.less中还有一些东西也可能类似。

**Reply by yoyo837** [reply]
试试`antd/dist/reset.css`

**Reply by zhangqhzz** [reply]
OK，谢谢

**Reply by glin7976** [reply]
你好，请问你解决了吗？我是了yoyo的方法不work

**Comment by zhangqhzz** [comment]
App包裹组件
如何使用部分（不是代码演示）， 能否也加入javascript，现在只有typescript的代码
https://ant.design/components/app-cn

**Reply by Wxh16144** [reply]
应该没有这方面计划，现在所有代码片段优先使用 `TypeScript`。

**Reply by afc163** [reply]
@PeachScript dumi 可以默认来个代码切换按钮？

**Reply by PeachScript** [reply]
自动切换比较难做，dumi 1 最初有对代码演示提供 ts 自动转 js 作转换，但构建时间会变长（compile + prettier）、产物尺寸也会变大（多一份 js），所以后面就去掉了

dumi 2 有计划支持 tab 形式，md 里一次性写两份代码块，dumi 再整合展示

**Reply by afc163** [reply]
可以做前端转换

**Comment by zhangqhzz** [comment]
希望在组件设计时，兼考虑主从、主从从等在一个画面操作的例子，另外定制主题能否有表单上预设的几种，如表头色，内容，奇偶行，选中色等。
![image](https://user-images.githubusercontent.com/10987820/219253528-4e87d645-6e82-44ce-870d-b5ba0c9b129c.png)
![image](https://user-images.githubusercontent.com/10987820/219254018-97e8819b-aaa6-4bee-b153-df484b4c059b.png)

**Reply by zhangqhzz** [reply]
pro的都比较大，需要滚动看另一项内容，否则一个画面上看表单行数就很少

**Comment by xiaogonggong-w** [comment]
<img width="619" alt="image" src="https://user-images.githubusercontent.com/79799040/219853255-9c65df14-b2a1-4725-b100-09c74f7c96c2.png">
按照升级文档来的,less,less-loader已经安装了,项目中并没有引入这个less文件,又重新删除了node_moudle包和.umi还是不行

**Reply by Jiangyidan** [reply]
解决了吗

**Reply by dx-pioneer** [reply]
<img width="421" alt="image" src="https://github.com/ant-design/ant-design/assets/75721352/2a6525f9-1ca6-49ae-ae50-63dabe44e01a">
这样配置一下，清除一下cache，重装node_modules和.umi，就可以了，我是这样做的

**Comment by jyw2001** [comment]
![image](https://user-images.githubusercontent.com/67536840/221513303-c2277a83-ff08-46f8-afa1-ac13fd565094.png)
二维码功能是已经没了么 还是说我依赖没有安装好？

**Reply by kiner-tang** [reply]
装的是什么版本的`antd`?`qrcode`是5.1.0之后才支持的哟
<img width="939" alt="image" src="https://user-images.githubusercontent.com/10286961/221514291-d926cfa8-e194-4f39-99fb-f1c908830402.png">

**Comment by sktony6260** [comment]
CompatibleConsumer.js:2 Uncaught TypeError: Cannot read properties of undefined (reading 'Consumer')

按照升级提示从v4升级到v5，浏览器里面有报错

错误文件是

webpack:///node_modules/@ant-design/compatible/es/CompatibleConsumer.js?1b55

错误代码

```
import { ConfigProvider } from 'antd';
var MergedConfigConsumer = ConfigProvider.ConfigContext.Consumer;
export default MergedConfigConsumer;
```
要怎么解决啊 @chenshuai2144

**Comment by hjane** [comment]
这是来自QQ邮箱的自动回复邮件。您好，您的邮件已经收到，我会在一周内尽快给您回复！

**Comment by hjane** [comment]
这是来自QQ邮箱的自动回复邮件。您好，您的邮件已经收到，我会在一周内尽快给您回复！

**Comment by hjane** [comment]
这是来自QQ邮箱的自动回复邮件。您好，您的邮件已经收到，我会在一周内尽快给您回复！

**Comment by wangjianjia1314** [comment]
umi3 antd4 在升级到antd5的时候出现两个错误
1.@import '~antd/es/style/themes/default.less';xx/.umi/plugin-layout/layout/layout' ；ant-design/pro-layout": "^7.21.2",
2.ERROR in ./src/.umi/core/routes.ts 9:0-69，Module not found: Error: Can't resolve '@ant-design/pro-layout/es/PageLoading' 
请问如何解决

**Comment by lanten** [comment]
`@ant-design/moment-webpack-plugin` 在 vite 中有替代方案吗？
